File: ANGEN/SampleFramework.Controls/FakeIconEditor/Samples/Snap UI Category/Fake IconEditor/CS/FakeIconEditor/MainForm.cs
using Genetibase.Shared.Controls;

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace FakeIconEditor
{
	public partial class MainForm : Form
	{
		protected override void OnFormClosed(FormClosedEventArgs e)
		{
			base.OnFormClosed(e);
			_blendForm.Dispose();
			_paletteForm.Dispose();
			NuGenUISnap.UnregisterExternalReferenceForm(this);
		}

		protected override void OnLoad(EventArgs e)
		{
			base.OnLoad(e);
			NuGenUISnap.RegisterExternalReferenceForm(this);
			_blendForm.Show();
			_paletteForm.Show();
		}

		private BlendForm _blendForm;
		private PaletteForm _paletteForm;

		public MainForm()
		{
			this.InitializeComponent();

			_blendForm = new BlendForm();
			_blendForm.Location = new Point(this.Right, this.Top);

			_paletteForm = new PaletteForm();
			_paletteForm.Location = new Point(this.Left - _paletteForm.Width, this.Top);
		}
	}
}

File: ANGEN/SampleFramework.Controls/NavigationBar/Samples/Snap UI Category/Fake IconEditor/VB/FakeIconEditor/MainForm.vb
Imports Genetibase.Shared.Controls

Friend NotInheritable Class MainForm
    Protected Overrides Sub OnFormClosed(ByVal e As System.Windows.Forms.FormClosedEventArgs)
        MyBase.OnFormClosed(e)

        RemoveHandler _blendForm.BlendColorChanged, New EventHandler(Of ColorEventArgs)(AddressOf _blendForm_BlendColorChanged)
        _blendForm.Dispose()
        RemoveHandler _paletteForm.GradientBeginChanged, New EventHandler(Of ColorEventArgs)(AddressOf _paletteForm_GradientBeginChanged)
        RemoveHandler _paletteForm.GradientEndChanged, New EventHandler(Of ColorEventArgs)(AddressOf _paletteForm_GradientEndChanged)
        _paletteForm.Dispose()
        NuGenUISnap.UnregisterExternalReferenceForm(Me)
    End Sub

    Protected Overrides Sub OnLoad(ByVal e As System.EventArgs)
        MyBase.OnLoad(e)
        NuGenUISnap.RegisterExternalReferenceForm(Me)
        _blendForm.Show()
        _paletteForm.Show()
    End Sub

    Private Sub _blendForm_BlendColorChanged(ByVal sender As Object, ByVal e As ColorEventArgs)
        _colorPanel.SelectedColor = e.Color
    End Sub

    Private Sub _paletteForm_GradientBeginChanged(ByVal sender As Object, ByVal e As ColorEventArgs)
        _blendForm.SetGradientBegin(e.Color)
    End Sub

    Private Sub _paletteForm_GradientEndChanged(ByVal sender As Object, ByVal e As ColorEventArgs)
        _blendForm.SetGradientEnd(e.Color)
    End Sub

    Dim _blendForm As BlendForm
    Dim _paletteForm As PaletteForm

    Sub New()
        Me.InitializeComponent()

        _blendForm = New BlendForm
        AddHandler Me._blendForm.BlendColorChanged, New EventHandler(Of ColorEventArgs)(AddressOf _blendForm_BlendColorChanged)
        _blendForm.Location = New Point(MyBase.Right, MyBase.Top)
        _paletteForm = New PaletteForm
        AddHandler _paletteForm.GradientBeginChanged, New EventHandler(Of ColorEventArgs)(AddressOf _paletteForm_GradientBeginChanged)
        AddHandler _paletteForm.GradientEndChanged, New EventHandler(Of ColorEventArgs)(AddressOf _paletteForm_GradientEndChanged)
        _paletteForm.Location = New Point((MyBase.Left - _paletteForm.Width), MyBase.Top)
    End Sub
End Class
